modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -774,11 +774,27 @@ def set_modality(appointment)
           modality = 'communityCare'
         end
 
-        Rails.logger.info("VAOS appointment id #{appointment[:id]} modality cannot be determined.") if modality.nil?
-
+        log_modality_failure(appointment) if modality.nil?
         appointment[:modality] = modality
       end
 
+      def log_modality_failure(appointment)
+        Rails.logger.info("VAOS appointment id #{appointment[:id]} modality cannot be determined.")
```

Comment:
How often is this issue occurring? Would it be better to log every error to get a better, more accurate gauge? Just wondering how the random logging was determined?

Reply:
We received feedback that this was triggering several millions of times per day so we were told to sample the errors to reduce the amount of logs we're collecting. Once we fix the underlying issue we'll remove the sampling.